siad/daemon_test.go

```diff
@@ -1,24 +1,144 @@
 package main
 
-func testingDaemon() (d *daemon, err error) {
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"testing"
+
+	"github.com/NebulousLabs/Sia/consensus"
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/network"
+)
+
+var (
+	APIPort int = 9020
+	RPCPort int = 9120
+)
+
+type daemonTester struct {
+	*daemon
+	*testing.T
+	rpcChan chan struct{}
+}
+
+func newDaemonTester(t *testing.T) *daemonTester {
 	dc := DaemonConfig{
-		APIAddr: ":9020",
-		RPCAddr: ":9021",
+		APIAddr: ":" + strconv.Itoa(APIPort),
+		RPCAddr: ":" + strconv.Itoa(RPCPort),
 
 		HostDir: "hostDir",
 
-		Threads: 2,
+		Threads: 1,
 
 		DownloadDir: "downloadDir",
 
-		WalletDir: "walletDir",
+		WalletDir: "walletDir" + strconv.Itoa(APIPort),
```

Comment:
this is pretty awful, it makes a bunch of walletDir directories. But it was necessary to prevent daemonTesters from reading from the same wallet file.